### PR TITLE
Run integration tests against C* 4.0

### PR DESCRIPTION
**Pull request by dimas-b**
Note: this is WIP

Note: C* 4.0 tests have schema agreement problems with more than one Stargate node, but even with 1 Stargate node (see failsafe config) `PersistenceTest` fails.

Running locally:

```
$ ./mvnw install -DskipTests
$ ccm create -n 1 -v 4.0-beta2 Test_Cluster
$ ccm start
$ ./mvnw -f testing/ failsafe:integration-test@cassandra-4.0 -Dmaven.failsafe.debug -Dstargate.test.backend.use.external=true
```

### CODE REVIEW

File: testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java [hunk 1]
```diff
@@ -321,6 +321,9 @@ public void startOsgi() {
                   + i
                   + ". The stargate node will not be started.",
               ex);
+
+          // If the container fails to star, there's no point running individual test cases.
```
Comment (dougwettlaufer) [inline testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java:325]: typo
```suggestion
          // If the container fails to start, there's no point running individual test cases.
```

**Comment by dougwettlaufer** [comment]
Are you working on ci.yml changes for this?

**Comment by dimas-b** [comment]
> Are you working on ci.yml changes for this?

I have it in the back of my mind, but just getting ITs to pass in plain maven env. is a challenge ATM :)

I suppose we can optimize CI later, once basic IT goals are passing.

**Comment by dimas-b** [comment]
`PersistenceTest` still failed in CI :(

File: persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java [hunk 2]
```diff
@@ -462,7 +463,16 @@ public boolean isInSchemaAgreement() {
     // Also note that in theory getSchemaVersion can return null for some nodes, and if it does
     // the code below will likely return false (the null will be an element on its own), but that's
     // probably the right answer in that case. In practice, this shouldn't be a problem though.
-    return Gossiper.instance.getLiveTokenOwners().stream()
+
+    // Important: This must include all nodes including fat clients, otherwise we'll get write
+    // errors
```
Comment (mpenick) [inline persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:468]: ugh, why does it format like this?

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java  1
- persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263917&bid=20396683)

File: persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java (hunk 2 above)
Comment (dimas-b) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:468]: over line length limit?

Comment (dimas-b) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:468]: try merging with the next line?

**Review by dougwettlaufer — APPROVED** [review]

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java  1
- persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263917&bid=20396683)

File: persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java (hunk 2 above)
Comment (mpenick) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:468]: It was, but the automatic formatter did this.

File: testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java [hunk 3]
```diff
@@ -84,7 +84,7 @@
 
   public static List<Starter> stargateStarters = new ArrayList<>();
   private static List<String> stargateHosts = new ArrayList<>();
-  public static final Integer numberOfStargateNodes = 3;
+  public static final Integer numberOfStargateNodes = Integer.getInteger("stargate.test.nodes", 3);
```
Comment (mpenick) [inline testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java:87]: Nice change.

**Pull request merged**